suite-common/wallet-core/src/accounts/accountsReducer.ts

```diff
@@ -86,15 +87,36 @@ export const prepareAccountsReducer = createReducerWithExtraDeps(
 
 export const selectAccounts = (state: AccountsRootState) => state.wallet.accounts;
 
-export const selectAccount = (state: AccountsRootState, accountKey: string) =>
-    state.wallet.accounts.find(account => account.key === accountKey);
+export const selectAccount = createSelector(
+    selectAccounts,
+    (_state: AccountsRootState, accountKey: string) => accountKey,
+    (accounts, accountKey) => accounts.find(account => account.key === accountKey),
+);
+
+export const selectDeviceNetworkAccounts = createSelector(
+    selectAccounts,
+    (_state: AccountsRootState, deviceState: string, networkSymbol: NetworkSymbol) => ({
+        deviceState,
+        networkSymbol,
+    }),
+    (accounts, params) =>
+        accounts.filter(
+            account =>
+                account.deviceState === params.deviceState &&
+                account.symbol === params.networkSymbol,
+        ),
+);
 
-// So far, mobile app doesn't persis data between app runs so until this is resolved
-// account names are just "Bitcon #<ACCOUNT_INDEX>"
 export const selectAccountName = createSelector(
     [selectAccount, selectAccounts],
     (account, accounts) => {
-        const accountIndex = accounts.findIndex(acc => acc.descriptor === account?.descriptor);
-        return `Bitcoin #${accountIndex + 1}`;
+        const accountData = accounts.find(acc => acc.descriptor === account?.descriptor);
+        if (accountData) {
+            const {
+                metadata: { accountLabel },
+            } = accountData;
+            return accountLabel;
+        }
+        return '';
```

Comment:
Return null so it screams in jsx and component can be modified to support no name?

Reply:
Yes, can do this - a9caefa2de0328a1277b97714a21a8a7c1969ab4